### ISSUE TITLE
Question on merge function usage

### ISSUE THREAD

**Issue by maldrake** [y-crdt/y-crdt#222]
Hi, I'm seeing something I didn't expect when using the `yrs::merge_updates_v1` function.  To better explain what I'm seeing, I put together a repository with a minimum reproduction of the issue: https://github.com/maldrake/yrs-minimum.  (If it's not minimal enough, let me know and I'll try to crunch it down farther.)

In summary, I have two yrs documents representing clients. Instead of synching with each other directly, peer-to-peer, they sync with a central server. In one test case, that server is modeled as a third `yrs::Doc` document, using `apply_update` to apply successive edits to the server state. This one seems to work fine. In the second test case, I change the server from using a document to keeping a `Vec<u8>` and using `merge_updates_v1` to merge in successive updates from the clients.  In this second case, it looks like I'm seeing the merged update that represents the server state taking on unexpected values or clearing our altogether.

For simplicity in the minimal reproduction of the issue, I left out any use of state vectors. I just used `StateVector::default()` everywhere. It's less efficient, but the idempotency of updates means it shouldn't affect correctness.

Is this a bug in the how updates are merged, or a misunderstanding on my part about how to use the API?

**Comment by zaynetro** [comment]
I haven't look deep into your sample repo but I use the following approach:

```rust
/// Store doc on disk or send to backend
pub fn encode_yrs_doc(doc: &yrs::Doc) -> Vec<u8> {
    let txn = doc.transact();
    txn.encode_state_as_update_v2(&StateVector::default())
}

/// Read doc from disk or from backend
pub fn apply_updates(yrs_client_id: yrs::block::ClientID, updates: &[u8]) -> Result<yrs::Doc> {
    let doc = yrs::Doc::with_client_id(yrs_client_id);
    let mut txn = source.transact_mut();
    let u = Update::decode_v2(updates)?;
    txn.apply_update(u);
    Ok(doc)
}
```

**Comment by maldrake** [comment]
@zaynetro,

Yes, thank you. There are two test cases in the repo. The first test case uses the method you described, instantiating a document and using `apply_update`. Everything works as expected in that case.  When using `yrs::merge_updates_v1(&[&update1, &update2])`, which is shown in the second test case, the behavior is not what I had expected.

It's possible that I just misunderstand the intended behavior and usage of `merge_updates_v1`.  If I am using it correctly, there may be a bug, so I thought I should open the issue here.  However, because the method of creating a `yrs::Doc` and applying updates to it works just fine, I'm certainly not blocked at all.

**Comment by maldrake** [comment]
It looks like there was a bug, and it was fixed by a recent PR.  When I updated the `yrs` version in `Cargo.toml` in the repo reproducing the problem to version 0.16.3 (released in #271), the problem was resolved.  I suspect this ended up being a duplicate of #268 and closed by #269.  I'm closing this issue, as it's no longer a problem.
